lib/matplotlib/ticker.py

```diff
@@ -2479,6 +2479,13 @@ def tick_values(self, vmin, vmax):
                   if mpl.rcParams['_internal.classic_mode'] else
                   (numdec + 1) // numticks + 1)
 
+        # if we have decided that the stride is as big or bigger than
```

Comment:
Is this really the right solution to the original problem?  To me, it seems that `numticks` is 1 because there isn't room for more ticks.  Do we really want ticks that overwrite each other for very narrow axes?

Reply:
If you are very narrow, there's no good solution. Either you have only one tick, which makes your scale uninterpretable. Or you have overlap.

Reply:
Sorry, let me rephrase: We clip numticks to 2, so why don't we have 2 ticks?

Reply:
`numticks` is an upper bound and only "best effort" at that.  

The issue is that based on `numtick` and a rounded "in" version of the range we pick a stride between major ticks (which is meant to catch things like 30 orders of magnitude with 9 ticks).  If we propose 3 ticks, have a range of [5, 150], then we "round out" to [1, 100] for the range giving us `numdec = 2` and `numticks = 3` and a stride of `(3 / 3) + 1 == 2` which means we put ticks at `(10 ^ {-2, 0, 2, 4})` which in turn means that only the tick at 100 is actually shown.  Additionally because we only show the minor ticks if the major stride is equal to 1 you end up with exactly 1 tick and 1 label shown even though the users / auto logic asked for up to 3.

I think that this fix (which fixes the values after doing a bunch of heuristics) is a better and more reliable fix than making the heuristics more complicated.